service_test.go

```diff
@@ -615,13 +615,13 @@ func (ds *DummyService) ProcessClientRequest(req *http.Request, path string, buf
 
 	if err := ds.c.RegisterProtocolInstance(dp); err != nil {
 		ds.link <- false
-		return nil, errors.New("")
+		return nil, nil, errors.New("")
```

Comment:
What is this errors.New("") about? (I know it was there before, but still seems strange.)

Reply:
I'm not sure, but it makes more sense to return `err`.